federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProvider.java

```diff
@@ -701,14 +713,17 @@ protected UserModel importUserFromLDAP(KeycloakSession session, RealmModel realm
                         .findFirst().orElse(null);
                 if (existingLocalUser != null) {
                     imported = existingLocalUser;
-                    // Need to evict the existing user from cache
+                    if (importType == ImportType.NOT_FORCED_RETURN_NULL) {
+                        // import not forced and return null
+                        return null;
+                    } else if (importType == ImportType.NOT_FORCED_RETURN_EXISTING) {
+                        // import not forced and return the current DB user
+                        return imported;
```

Comment:
One comment here @sguilhen, maybe we should do here the `proxy(realm, imported, ldapUser, false)` as we are doing in other parts. WDYT?

Reply:
That is probably a good idea, yes.

Reply:
Done, updated the first commit to proxy the local user